subprojects/file-collections/src/main/java/org/gradle/api/internal/file/collections/FilteredMinimalFileTree.java

```diff
@@ -74,12 +74,7 @@ public VisitType prepareForVisit(FileCollectionInternal.Source source) {
 
             @Override
             public void visitCollection(FileCollectionInternal.Source source, Iterable<File> contents) {
-                visitor.visitGenericFileTree(owner, FilteredMinimalFileTree.this);
-            }
-
-            @Override
-            public void visitGenericFileTree(FileTreeInternal fileTree, FileSystemMirroringFileTree sourceTree) {
-                visitor.visitGenericFileTree(owner, FilteredMinimalFileTree.this);
+                visitor.visitCollection(source, contents);
```

Comment:
This is the only place where I am not sure what the effect is. I think this means that if the `MinimalFileTree` says it is a collection, then we'd pass it through. Which seems to make sense.

Reply:
This comment is now outdated, has this been resolved already?

Reply:
This is gone now since the only usage was in the old continuous build code.